sympy/holonomic/holonomic.py

```diff
@@ -520,7 +520,7 @@ def _singularics_to_ord(self):
             y0 = []
             for i in range(a):
                 y0.append(S(0))
-            y0 += b
+            y0 += [j * factorial(a + i) for i, j in enumerate(b)]
```

Comment:
And maybe it should even be tested that `int(a)` is still an exact root of the indicial equation, not just an approximate root.